Sources/MongoSwift/Operations/ListIndexesOperation.swift

```diff
@@ -0,0 +1,51 @@
+import mongoc
+
+/// Internal intermediate result of a ListIndexes command.
+internal enum ListIndexesResults {
+    /// Includes the name, namespace, version, keys, and options of each index.
+    case specs(MongoCursor<IndexModel>)
+
+    /// Only includes the names.
+    case names([String])
+}
+
+/// An operation corresponding to a listIndex command on a collection.
+internal struct ListIndexesOperation<T: Codable>: Operation {
+    private let collection: MongoCollection<T>
+    private let nameOnly: Bool
+
+    internal init(collection: MongoCollection<T>, nameOnly: Bool) {
+        self.collection = collection
+        self.nameOnly = nameOnly
+    }
+
+    internal func execute(using connection: Connection, session: ClientSession?) throws -> ListIndexesResults {
+        let opts = try encodeOptions(options: Document(), session: session)
+
+        let initializer = { (conn: Connection) -> OpaquePointer in
+            self.collection.withMongocCollection(from: conn) { collPtr in
+                guard let indexes = mongoc_collection_find_indexes_with_opts(collPtr, opts?._bson) else {
+                    fatalError(failedToRetrieveCursorMessage)
+                }
+                return indexes
+            }
+        }
+        if self.nameOnly {
```

Comment:
because there isn't a server-side `nameOnly` parameter, I think we can just have this operation always return a cursor over the specifications, and we can map that cursor to a `[String]` in the body of `listCollectionNames`.

The separation in `ListCollectionOperation` and `ListDatabaseOperation` is mainly due to the fact that `nameOnly` affects both the format of the server's replies as well as the semantics of the command itself. In this case, the server returns the same thing regardless of whether we only want the names or not, so I think the operation should do the same.